routing/localchans/manager_test.go

```diff
@@ -115,34 +133,162 @@ func TestManager(t *testing.T) {
 		}, nil
 	}
 
+	isChannelActive := func(chanID lnwire.ChannelID) bool {
+		for _, c := range channelSet {
+			if !chanID.IsChanPoint(&c.edgeInfo.ChannelPoint) {
+				continue
+			}
+
+			return c.active
+		}
+		return false
+	}
+
 	manager := Manager{
 		UpdateForwardingPolicies:  updateForwardingPolicies,
 		PropagateChanPolicyUpdate: propagateChanPolicyUpdate,
 		ForAllOutgoingChannels:    forAllOutgoingChannels,
 		FetchChannel:              fetchChannel,
+		IsChannelActive:           isChannelActive,
 	}
 
-	// Test updating a specific channels.
-	err := manager.UpdatePolicy(newPolicy, chanPoint)
-	if err != nil {
-		t.Fatal(err)
-	}
+	// Policy with no max htlc value.
+	MaxHTLCPolicy := currentPolicy
+	MaxHTLCPolicy.MaxHTLC = newPolicy.MaxHTLC
+	noMaxHtlcPolicy := newPolicy
+	noMaxHtlcPolicy.MaxHTLC = 0
 
-	// Test updating all channels, which comes down to the same as testing a
-	// specific channel because there is only one channel.
-	err = manager.UpdatePolicy(newPolicy)
-	if err != nil {
-		t.Fatal(err)
+	tests := []struct {
+		name                     string
+		currentPolicy            channeldb.ChannelEdgePolicy
+		newPolicy                routing.ChannelPolicy
+		channelSet               []channel
+		specifiedChanPoints      []wire.OutPoint
+		expectedNumUpdates       int
+		expectedNumFailedUpdates int
```

Comment:
Can we match these to an exact set of failure updates we expect?

Reply:
Sure thing. Np.